### PR TITLE
Remove /teams page

### PR DESCRIPTION
**Pull request by danieljames-dj**
The page worldcubeassociation.org/teams is currently not used by anybody. I was thinking that this page will be used to access Team.banned. But just found that banned competitors are actually accessed from https://www.worldcubeassociation.org/disciplinary. Hence I thought removing this page will be good. Also, this will help to avoid leaders accidentally going to this page (using some bookmarked URL) instead of the new page.

### CODE REVIEW

File: app/controllers/teams_controller.rb [hunk 1]
```diff
@@ -2,13 +2,8 @@
 
 class TeamsController < ApplicationController
   before_action :authenticate_user!
-  before_action -> { redirect_to_root_unless_user(:can_manage_teams?) }, except: [:edit, :update]
```
Comment (gregorbg) [inline app/controllers/teams_controller.rb:5]: I think this safeguard should stay in place, with an adequate replacement of `can_manage_teams?`

File: app/webpacker/lib/requests/routes.js.erb [hunk 2]
```diff
@@ -147,8 +147,6 @@ export const subordinateDelegateClaimsUrl = `<%= CGI.unescape(Rails.application.
 export const subordinateUpcomingCompetitionsUrl = `<%= CGI.unescape(Rails.application.routes.url_helpers.competitions_for_senior_path) %>`;
 export const wfcCompetitionsExportUrl = `<%= CGI.unescape(Rails.application.routes.url_helpers.wfc_competitions_export_path) %>`;
 
-export const teamUrl = (id) => `<%= CGI.unescape(Rails.application.routes.url_helpers.team_path("${id}"))%>`;
```
Comment (gregorbg) [inline app/webpacker/lib/requests/routes.js.erb:150]: Have you made sure that this isn't used across our webpacker code?

**Review by gregorbg — CHANGES_REQUESTED** [review]

File: app/webpacker/lib/requests/routes.js.erb (hunk 2 above)
Comment (danieljames-dj) [reply app/webpacker/lib/requests/routes.js.erb:150]: Yes, it's not used anywhere in our codebase.

File: app/controllers/teams_controller.rb (hunk 1 above)
Comment (danieljames-dj) [reply app/controllers/teams_controller.rb:5]: This safeguard was for `def index`, but the `def index` won't exist after this PR is merged, that is the reason I removed it. There is already safeguard for other actions in the next line.

Comment (gregorbg) [reply app/controllers/teams_controller.rb:5]: Ah okay, in that case I misread something in the controller. Thanks for confirming!

**Review by gregorbg — APPROVED** [review]

**Pull request merged**